app/src/main/java/ch/hikemate/app/ui/map/RunHikeScreen.kt

```diff
@@ -0,0 +1,172 @@
+
+  // Only do the configuration on the first composition, not on every recomposition
+  LaunchedEffect(Unit) {
+    Configuration.getInstance().apply {
+      // Set user-agent to avoid rejected requests
+      userAgentValue = context.packageName
+
+      // Allow for faster loading of tiles. Default OSMDroid value is 2.
+      tileDownloadThreads = 4
+
+      // Maximum number of tiles that can be downloaded at once. Default is 40.
+      tileDownloadMaxQueueSize = 40
+
+      // Maximum number of bytes that can be used by the tile file system cache. Default is 600MB.
+      tileFileSystemCacheMaxBytes = 600L * 1024L * 1024L
+    }
+  }
+
+  // Avoid re-creating the MapView on every recomposition
+  val mapView = remember {
+    MapView(context).apply {
+      controller.setZoom(routeZoomLevel)
+      controller.setCenter(MapUtils.getGeographicalCenter(route.bounds))
+      // Limit the zoom to avoid the user zooming out or out too much
+      minZoomLevel = routeZoomLevel
+      // Avoid repeating the map when the user reaches the edge or zooms out
+      // We keep the horizontal repetition enabled to allow the user to scroll the map
+      // horizontally without limits (from Asia to America, for example)
+      isHorizontalMapRepetitionEnabled = true
+      isVerticalMapRepetitionEnabled = false
+      // Disable built-in zoom controls since we have our own
+      zoomController.setVisibility(CustomZoomButtonsController.Visibility.NEVER)
+      // Enable touch-controls such as pinch to zoom
+      setMultiTouchControls(true)
+    }
+  }
+
+  // When the map is ready, it will have computed its bounding box
+  mapView.addOnFirstLayoutListener { _, _, _, _, _ ->
+    // Limit the vertical scrollable area to avoid the user scrolling too far from the hike
+    mapView.setScrollableAreaLimitLatitude(
+        min(MapScreen.MAP_MAX_LATITUDE, mapView.boundingBox.latNorth),
+        max(MapScreen.MAP_MIN_LATITUDE, mapView.boundingBox.latSouth),
+        HikeDetailScreen.MAP_BOUNDS_MARGIN)
+    if (route.bounds.maxLon < HikeDetailScreen.MAP_MAX_LONGITUDE ||
+        route.bounds.minLon > HikeDetailScreen.MAP_MIN_LONGITUDE) {
+      mapView.setScrollableAreaLimitLongitude(
+          max(HikeDetailScreen.MAP_MIN_LONGITUDE, mapView.boundingBox.lonWest),
+          min(HikeDetailScreen.MAP_MAX_LONGITUDE, mapView.boundingBox.lonEast),
+          HikeDetailScreen.MAP_BOUNDS_MARGIN)
+    }
+  }
+
+  val hikeLineColor = route.getColor()
+  MapUtils.showHikeOnMap(mapView = mapView, hike = route, color = hikeLineColor, onLineClick = {})
+
+  val errorMessageIdState = profileViewModel.errorMessageId.collectAsState()
+  val profileState = profileViewModel.profile.collectAsState()
+
+  AsyncStateHandler(
+      errorMessageIdState = errorMessageIdState,
+      actionContentDescriptionStringId = R.string.go_back,
+      actionOnErrorAction = { navigationActions.navigateTo(Route.MAP) },
```

Comment:
Fix: I think it would be better to go back to the previous screen in an error case.